### PR TITLE
adding Memory methods (#138)

### LINKED ISSUE #138: Add Methods to  the client to affect VM Memory
needed methods:

WithMemory(MemoryMB int64) (BuildableVMParameters, error)
MustWithMemory(MemoryMB int64) BuildableVMParameters

### PR DESCRIPTION
**Pull request by eslutsky**
## Please describe the change you are making

This PR fixes #138 and adds the memory field to the VM object.

## Are you the owner of the code you are sending in, or do you have permission of the owner?

Yes

## The code will be published under the BSD 3 clause license. Have you read and understood this license?

Yes

### CODE REVIEW

File: client_vm_create.go [hunk 1]
```diff
@@ -45,6 +45,13 @@ func vmBuilderHugePages(params OptionalVMParameters, builder *ovirtsdk.VmBuilder
 	}
 }
 
+func vmBuilderMemory(params OptionalVMParameters, builder *ovirtsdk.VmBuilder) {
+	//panic(newError(EBug, "Failed to build 'hugepages' custom property from value %d", params.Memory()))
```
Comment (janosdebugs) [inline client_vm_create.go:49]: Please don't commit commented-out code.

File: client_vm.go [hunk 2]
```diff
@@ -579,19 +587,36 @@ func CreateVMParams() BuildableVMParameters {
...
-	comment string
-	cpu     VMCPUTopo
-
+	name      string
+	comment   string
+	cpu       VMCPUTopo
 	hugePages *VMHugePages
 
 	initialization Initialization
+	memory         int64
```
Comment (janosdebugs) [inline client_vm.go:596]: Shouldn't this be `*int64` so you can indicate a `nil` value? That would help with determining if the memory has been set or not.

File: client_vm_test.go [hunk 3]
```diff
@@ -139,6 +139,30 @@ func TestVMCreationWithCPU(t *testing.T) {
 	}
 }
 
+func TestVMCreationWithMemory(t *testing.T) {
+
+	params := map[string]ovirtclient.OptionalVMParameters{
+		"withmemory": ovirtclient.CreateVMParams().MustWithMemory(1048576),
+	}
+	for name, param := range params {
```
Comment (janosdebugs) [inline client_vm_test.go:147]: Please keep the tests simple and  readable, I don't think we need the fancy data-driven logic here.

**Review by janosdebugs — CHANGES_REQUESTED** [review]
Looks good, a few changes, and the mocks are still missing.

File: client_vm_create.go (hunk 1 above)
Comment (eslutsky) [reply client_vm_create.go]: fixed

File: client_vm.go (hunk 2 above)
Comment (eslutsky) [reply client_vm.go]: fixed

File: client_vm_test.go (hunk 3 above)
Comment (eslutsky) [reply client_vm_test.go]: fixed

File: client_vm.go [hunk 4]
```diff
@@ -706,6 +731,7 @@ type vm struct {
 	templateID     TemplateID
 	status         VMStatus
 	cpu            *vmCPU
+	memory         *int64
```
Comment (janosdebugs) [inline client_vm.go:734]: Why is this a pointer? It seems to me that the API always returns the memory field.

File: mock_vm_create.go [hunk 5]
```diff
@@ -79,6 +79,7 @@ func (m *mockClient) createVM(
 		status:         VMStatusDown,
 		cpu:            cpu,
 		hugePages:      params.HugePages(),
+		memory:         params.Memory(),
```
Comment (janosdebugs) [inline mock_vm_create.go:82]: Please handle the case when this is not set. What is the default?

File: client_vm_test.go [hunk 6]
```diff
@@ -139,6 +139,24 @@ func TestVMCreationWithCPU(t *testing.T) {
 	}
 }
 
+func TestVMCreationWithMemory(t *testing.T) {
```
Comment (janosdebugs) [inline client_vm_test.go:142]: Please add a test that checks the default behavior if the memory is not set. How much memory is allocated by default?

**Review by janosdebugs — CHANGES_REQUESTED** [review]

File: client_vm.go (hunk 4 above)
Comment (eslutsky) [reply client_vm.go]: Fixed

File: mock_vm_create.go (hunk 5 above)
Comment (eslutsky) [reply mock_vm_create.go]: Fixed

File: client_vm_test.go (hunk 6 above)
Comment (eslutsky) [reply client_vm_test.go:142]: Fixed

File: mock_vm_create.go [hunk 7]
```diff
@@ -69,6 +69,11 @@ func (m *mockClient) createVM(
 	if init == nil {
 		init = &initialization{}
 	}
+	memory := int64(1073741824)
```
Comment (janosdebugs) [inline mock_vm_create.go:72]: Is this the same default as in the oVirt engine?

**Review by janosdebugs — APPROVED** [review]

File: mock_vm_create.go (hunk 7 above)
Comment (eslutsky) [reply mock_vm_create.go:72]: yes

**Pull request merged**